src/incremental/update.py

```diff
@@ -6,9 +6,51 @@
 import click
 import os
 import datetime
+from typing import TYPE_CHECKING, Dict, Optional, Callable, Union, Iterable
 
 from incremental import Version
-from twisted.python.filepath import FilePath
+
+if TYPE_CHECKING:
+    from typing_extensions import Protocol
+
+    class _ReadableWritable(Protocol):
+        def read(self):  # type: () -> bytes
+            pass
+
+        def write(self, v):  # type: (bytes) -> object
+            pass
+
+        def __enter__(self):  # type: () -> _ReadableWritable
+            pass
+
+        def __exit__(self, *args, **kwargs):  # type: (object, object) -> Optional[bool]
+            pass
+
+    class FilePath(object):
```

Comment:
FilePath is type annotated in twisted, but it's decorated with `twisted.python.comparable` which strips the annotations